packages/@cdktf/hcl2cdk/lib/schema.ts

```diff
@@ -0,0 +1,43 @@
+import * as z from "zod";
+import { ZodRawShape } from "zod/lib/src/types/base";
+
+const tfObject = <T extends ZodRawShape>(config: T) =>
+  z.array(z.object(config).partial());
+
+const outputConfig = tfObject({
+  value: z.string(),
+  description: z.string().optional(),
+  sensitive: z.boolean().optional(),
+  depends_on: z.string().optional(),
+});
+export type Output = z.infer<typeof outputConfig>;
+
+const variableConfig = tfObject({
+  type: z.string(),
+  default: z.any(),
+  description: z.string(),
+  sensitive: z.boolean(),
+});
+export type Variable = z.infer<typeof variableConfig>;
+
+const providerConfig = z.array(z.record(z.any()));
+export type Provider = z.infer<typeof providerConfig>;
+
+const moduleConfig = z.array(z.object({ source: z.string() }).nonstrict());
+export type Module = z.infer<typeof moduleConfig>;
+
+const resourceConfig = z.array(z.record(z.any()));
+export type Resource = z.infer<typeof resourceConfig>;
+export type Data = Resource;
+
+export const schema = z
+  .object({
```

Comment:
What about the [Terraform block](https://www.terraform.io/docs/language/syntax/json.html#terraform-blocks)?

Reply:
That we will need for sure! I'm adding this when I'm working on the import command, we need the data to construct a complete cdktf.json modules / providers list